# Day 5 – Pandas & Basic Dataset Analysis

**Intern:** Shri Sanjaykumar V  
**Role:** AI/ML Intern  
**Date:** 02 September 2026  

This notebook covers foundational dataset handling using **Pandas**, including DataFrames, inspecting rows/columns, checking missing values, filtering data, and computing summary statistics.

## 1. Introduction to Pandas

**Pandas** is the primary data manipulation and analysis library in Python. It provides fast, flexible data structures—most notably the **DataFrame**—designed to make working with structured (tabular, CSV, SQL) data intuitive and efficient for machine learning workflows.

## 2. Importing Pandas & Version Verification

In [1]:
import pandas as pd

print("Pandas successfully imported. Version:", pd.__version__)

Pandas successfully imported. Version: 3.0.3


## 3. Loading a CSV Dataset
Using `pd.read_csv()` to load our tabular dataset (`dataset.csv`) into a Pandas DataFrame.

In [2]:
df = pd.read_csv("dataset.csv")
print("Dataset loaded successfully!")

Dataset loaded successfully!


## 4. Understanding the DataFrame
A **DataFrame** is a two-dimensional labeled data structure with columns of potentially different types (like a spreadsheet or SQL table).

In [3]:
print("Data Structure Type:", type(df))
print("Columns:", df.columns)

Data Structure Type: <class 'pandas.core.frame.DataFrame'>
Columns: Index(['Employee_ID', 'Name', 'Department', 'Age', 'Salary',
       'Experience_Years'],
      dtype='object')


## 5. Displaying First 5 Rows
Using `df.head()` to inspect the top 5 records.

In [4]:
df.head()

   Employee_ID          Name    Department  Age   Salary  Experience_Years
0          101  Aarav Sharma   Engineering   28  72000.0               4.0
1          102    Diya Patel     Marketing   24  52000.0               2.0
2          103   Rohan Verma   Engineering   32  88000.0               7.0
3          104   Ananya Iyer  Data Science   26  78000.0               3.0
4          105   Kavya Reddy     Marketing   29      NaN               5.0


## 6. Displaying Last 5 Rows
Using `df.tail()` to inspect the bottom 5 records.

In [5]:
df.tail()

   Employee_ID           Name    Department  Age   Salary  Experience_Years
5          106   Vikram Singh  Data Science   35  95000.0              10.0
6          107     Neha Joshi   Engineering   25  60000.0               2.0
7          108     Rahul Nair            HR   30  55000.0               NaN
8          109    Pooja Hegde  Data Science   27  82000.0               4.0
9          110  Siddharth Das            HR   31  58000.0               6.0


## 7. Checking Rows and Columns
Using `df.shape` to check the dataset dimensions `(rows, columns)`.

In [6]:
print("Dataset Shape:", df.shape)
print(f"Total Rows: {df.shape[0]}, Total Columns: {df.shape[1]}")

Dataset Shape: (10, 6)
Total Rows: 10, Total Columns: 6


## 8. Checking Missing Values
Using `df.isnull().sum()` to count missing (null/NaN) entries per column.

In [7]:
print("Missing values per column:")
print(df.isnull().sum())

Missing values per column:
Employee_ID         0
Name                0
Department          0
Age                 0
Salary              1
Experience_Years    1


## 9. Filtering Data Using Conditions
Filtering records based on conditional logic.

In [8]:
# Condition 1: High earners (Salary > 70,000)
print("--- Condition 1: Employees with Salary > 70,000 ---")
print(df[df["Salary"] > 70000])

# Condition 2: Specific department (Department == 'Data Science')
print("\n--- Condition 2: Employees in Data Science Department ---")
print(df[df["Department"] == "Data Science"])

--- Condition 1: Employees with Salary > 70,000 ---
   Employee_ID          Name    Department  Age   Salary  Experience_Years
0          101  Aarav Sharma   Engineering   28  72000.0               4.0
2          103   Rohan Verma   Engineering   32  88000.0               7.0
3          104   Ananya Iyer  Data Science   26  78000.0               3.0
5          106  Vikram Singh  Data Science   35  95000.0              10.0
8          109   Pooja Hegde  Data Science   27  82000.0               4.0

--- Condition 2: Employees in Data Science Department ---
   Employee_ID          Name    Department  Age   Salary  Experience_Years
3          104   Ananya Iyer  Data Science   26  78000.0               3.0
5          106  Vikram Singh  Data Science   35  95000.0              10.0
8          109   Pooja Hegde  Data Science   27  82000.0               4.0


## 10. Summary Statistics
Using `df.describe()` to compute statistical metrics (count, mean, std, min, 25%, 50%, 75%, max) for numeric columns.

In [9]:
df.describe()

       Employee_ID       Age        Salary  Experience_Years
count     10.00000  10.00000      9.000000          9.000000
mean     105.50000  28.70000  71111.111111          4.777778
std        3.02765   3.40098  15584.001768          2.587362
min      101.00000  24.00000  52000.000000          2.000000
25%      103.25000  26.25000  58000.000000          3.000000
50%      105.50000  28.50000  72000.000000          4.000000
75%      107.75000  30.75000  82000.000000          6.000000
max      110.00000  35.00000  95000.000000         10.000000


## 11. Basic Observations

Based directly on the actual dataset results above:
1. **Dataset Dimensions:** The dataset has 10 rows and 6 columns (`Employee_ID`, `Name`, `Department`, `Age`, `Salary`, `Experience_Years`).
2. **Missing Values:** There are 2 missing values—1 in `Salary` (Row 4, Kavya Reddy) and 1 in `Experience_Years` (Row 7, Rahul Nair).
3. **Salary Distribution:** Salaried employees earn between 52,000 and 95,000, with an average salary of approximately 71,111.
4. **High Earners:** Exactly 5 employees earn more than 70,000.
5. **Department Breakdown:** There are 3 employees working in Data Science, 3 in Engineering, 2 in Marketing, and 2 in HR.
6. **Age & Experience:** The average employee age is ~28.7 years (range: 24 to 35), and average experience is ~4.8 years (range: 2 to 10 years).

## 12. Conclusion

In Day 5:
- Learned to load and inspect CSV datasets using Pandas DataFrames.
- Used `head()` and `tail()` to examine sample rows.
- Checked dataset dimensionality with `shape`.
- Identified missing data with `isnull().sum()`.
- Filtered tabular data using relational conditions.
- Generated and interpreted numerical summary statistics using `describe()`.